apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-inserter-menu-search-term.js

```diff
@@ -42,11 +43,33 @@ const InserterMenuTrackingEvent = function () {
 		tracksRecordEvent( 'wpcom_block_picker_no_results', eventProperties );
 	}, 500 );
 
+	const debouncedSetFilterValueDiscovery = debounce( ( search_term ) => {
+		if ( search_term.length < 3 ) {
+			return;
+		}
+
+		const blockDirectoryResults = document.querySelectorAll(
+			'.block-directory-downloadable-block-list-item'
+		).length;
+
+		if ( blockDirectoryResults ) {
+			tracksRecordEvent( 'wpcom_block_discovery_results', {
+				search_term,
+				results: blockDirectoryResults,
+			} );
+		} else {
+			tracksRecordEvent( 'wpcom_block_discovery_no_results', {
+				search_term,
+			} );
```

Comment:
Will this always record an event on sites with a Free plan? It'd be good if we only record the `no_results` event when we know a block discovery query was actually performed.

Would searching for the presence of `.block-directory-downloadable-blocks-panel` be enough?

Reply:
Thanks @p-jackson. I used your suggestion and check for `.block-directory-downloadable-blocks-panel` and `.block-editor-inserter__tips` because after having results, it show a tip element instead of the first one